### PR TITLE
Image classification/auto detect column type

### PR DESCRIPTION
**Pull request by abhishekpratapa**
- auto detect annotation type.

### CODE REVIEW

File: src/unity/lib/annotation/utils.cpp [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#include "utils.hpp"
```
Comment (TobyRoseman) [inline src/unity/lib/annotation/utils.cpp]: copyright header

File: src/unity/lib/annotation/annotation_base.hpp [hunk 2]
```diff
@@ -20,7 +20,7 @@
 #include "build/format/cpp/message.pb.h"
 #include "build/format/cpp/meta.pb.h"
 
-namespace annotate_spec = TuriCreate::Annotation::Specification;
```
Comment (TobyRoseman) [inline src/unity/lib/annotation/annotation_base.hpp]: This file doesn't have a copyright header

File: src/unity/lib/annotation/utils.hpp [hunk 3]
```diff
@@ -6,6 +6,10 @@
 #include "build/format/cpp/message.pb.h"
 #include "build/format/cpp/meta.pb.h"
 
+#include <boost/regex.hpp>
```
Comment (TobyRoseman) [inline src/unity/lib/annotation/utils.hpp]: Please check to make sure all of your files have a copyright header

File: src/unity/lib/annotation/utils.hpp [hunk 4]
```diff
@@ -28,6 +32,9 @@ populate_parcel(annotate_spec::Parcel &parcel, T message) {
   parcel.mutable_metadata()->CopyFrom(message);
 }
 
+bool isInteger(std::string s);
```
Comment (TobyRoseman) [inline src/unity/lib/annotation/utils.hpp]: `std::string` -> `const std::string`

File: src/unity/lib/annotation/image_classification.cpp [hunk 5]
```diff
@@ -234,6 +236,54 @@ void ImageClassification::_addAnnotationToSFrame(size_t index, int label) {
...
+          m_data->select_column(m_annotation_column));
+
+    std::cout << m_data->dtype() << std::endl;
+
+    std::shared_ptr<unity_sarray_base> integer_annotations =
+        data_sarray->astype(flex_type_enum::INTEGER, true);
+
+    m_data->remove_column(annotation_column_index);
+    m_data->add_column(integer_annotations, m_annotation_column);
+    std::cout << m_data->dtype() << std::endl;
```
Comment (znation) [inline src/unity/lib/annotation/image_classification.cpp]: Remove this line

File: src/unity/lib/annotation/image_classification.cpp [hunk 6]
```diff
@@ -234,6 +236,54 @@ void ImageClassification::_addAnnotationToSFrame(size_t index, int label) {
...
+      copy_data->drop_missing_values(annotation_column_name, false, false);
+
+  std::shared_ptr<unity_sframe> filtered_sframe =
+      std::static_pointer_cast<unity_sframe>(dropped_missing.front());
+
+  std::shared_ptr<unity_sarray> data_sarray =
+      std::static_pointer_cast<unity_sarray>(
+          filtered_sframe->select_column(m_annotation_column));
+
+  std::vector<flexible_type> flex_data = data_sarray->to_vector();
```
Comment (znation) [inline src/unity/lib/annotation/image_classification.cpp]: Should not need the cast to vector; could loop over the SArray data directly.

Comment (znation) [reply src/unity/lib/annotation/image_classification.cpp]: Does not need to be fixed in this PR.

File: src/unity/lib/annotation/image_classification.cpp [hunk 7]
```diff
@@ -234,6 +236,54 @@ void ImageClassification::_addAnnotationToSFrame(size_t index, int label) {
...
+      break;
+    }
+  }
+
+  if(castable){
+    std::shared_ptr<unity_sarray> data_sarray =
+      std::static_pointer_cast<unity_sarray>(
+          m_data->select_column(m_annotation_column));
+
+    std::cout << m_data->dtype() << std::endl;
```
Comment (znation) [inline src/unity/lib/annotation/image_classification.cpp]: Remove this line

File: src/unity/lib/annotation/image_classification.hpp [hunk 8]
```diff
@@ -28,6 +28,8 @@ class ImageClassification : public AnnotationBase {
 
   bool setAnnotations(const annotate_spec::Annotations &annotations) override;
 
+  void cast_annotations() override;
```
Comment (znation) [inline src/unity/lib/annotation/image_classification.hpp]: Should have `virtual`

Comment (znation) [reply src/unity/lib/annotation/image_classification.hpp]: Nevermind, StackOverflow agrees with you; this is fine as is.

**Comment by abhishekpratapa** [comment]
fixes #1540

**Pull request merged**